awx/sso/backends.py

```diff
@@ -473,16 +476,21 @@ def on_populate_user(sender, **kwargs):
 
 
 def reconcile_users_org_team_mappings(user, desired_org_states, desired_team_states, source):
+    #
+    # desired_org_states: { '<org_name>': { '<role>': <boolean> } }
+    #
+    # desired_team_states: { '<org_name>': { '<team name>': { '<role>': <boolean> } } }
```

Comment:
Really really helpful to have here, and just to capture as a TODO item, I suggest changing `<boolean>` to `<boolean> or None` for type clarity. Although it looks like the caller could pop off all None values to the same effect.